crates/primitives/src/transaction/mod.rs

```diff
@@ -943,6 +946,47 @@ impl Compact for TransactionSignedNoHash {
     }
 }
 
+#[cfg(not(feature = "zstd-codec"))]
+impl Compact for TransactionSignedNoHash {
```

Comment:
This is a new code right? Let's provide a unit test as well.

Reply:
Yeah, it is reduced version of "full" implementation, which is defined above.

For the unit test, I don't know how to make it run with all other tests, because it touches code behind the feature.